packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -371,6 +376,48 @@ export class AndroidEmulatorDevice extends DeviceBase {
     ]);
   }
 
+  async mirrorNativeLogsDispose() {
+    this.nativeLogsOutputChannel?.dispose();
+    this.nativeLogsProcess?.kill();
+  }
+
+  async mirrorNativeLogs(build: AndroidBuildResult) {
+    const startTime = strftime("%F %T.000", new Date());
+
+    const extractPidFromLogcat = async () =>
+      new Promise<string>((resolve, reject) => {
+        const process = exec(ADB_PATH, ["logcat"]);
```

Comment:
this process can run indefinitely in case there's some issue launching the app.

We should likely adapt schema that we use in couple of different places in the codebase and use `cancelToken` here instead of retaining only the the final logcat process.

What we should do is to have cancelToken that we'd use in dispose. Cancelling it would kill either the first or the second logcat process.

Also, would it make sense to filter the logs by passing some options like `-e` and also limiting the output with `-t 0` such that we have fewer lines to process? logcat can sometimes generate a large volumes of output and as long as we limit it when filtering by PID in the next command, we're not doing it here.

Reply:
I didn't use `-t` param, as it looks like it is not counting by the filtered messages. Instead, I added setTimeout for this method to reject if it won't find the pid in 2s. Changed in: https://github.com/software-mansion/radon-ide/pull/734/commits/bfa2d8cf60e5d5278b4150daee0dd02cb93e3778